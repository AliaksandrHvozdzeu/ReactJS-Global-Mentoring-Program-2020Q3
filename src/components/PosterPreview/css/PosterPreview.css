.poster-preview-overlay {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: rgb(0 0 0 / 96%);
    z-index: 2;
}

.poster-preview-body {
    z-index: 20;
    position: absolute;
    visibility: visible;
    background: #232323;
}

.congratulation-poster {
    width: 25rem;
    height: 600px;
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    background: #232323;
    overflow: auto;
    outline: none;
    padding: 52px 52px 30px;
}

.preview-poster-image {
    position: absolute;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
    width: 400px;
    height: 600px;
}
