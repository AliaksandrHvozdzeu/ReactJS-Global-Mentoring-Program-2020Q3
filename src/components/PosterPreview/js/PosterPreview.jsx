import React from 'react';
import PropTypes from 'prop-types';
import CloseButton from '../../CloseButton';
import useDefaultPoster from '../../../hooks/useDefaultPoster';
import '../css/PosterPreview.css';

export default function PosterPreview({ isOpen, onClose, src, alt, fallback }) {

  const className = 'preview-poster-image';

  const previewImage = useDefaultPoster(src, alt, className, fallback);

  return (
    <>
      {isOpen && (
        <div className="poster-preview-overlay">
          <div className="poster-preview-body">
            <section className="congratulation-poster">
              <CloseButton closeAction={onClose}/>
              {previewImage}
            </section>
          </div>
        </div>
      )}
    </>
  );


}

PosterPreview.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  fallback: PropTypes.string.isRequired,
};
