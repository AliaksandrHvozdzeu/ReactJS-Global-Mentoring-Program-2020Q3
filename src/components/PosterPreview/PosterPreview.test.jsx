import React from "react";
import rerender from "react-test-renderer";
import Constants from '../../constants';
import PosterPreview from "./js/PosterPreview";

describe("when PosterPreview", () => {

  const isOpen = true;
  const onClose = () => {
  };
  const src = Constants.DEFAULT_POSTER;
  const alt = "movie poster logo";
  const fallback = Constants.DEFAULT_POSTER;

  test("then snapshot created", () => {
    const component = rerender.create(<PosterPreview isOpen={isOpen}
                                                     onClose={onClose}
                                                     src={src}
                                                     alt={alt}
                                                     fallback={fallback} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("when scr is not default", () => {
    const src2 = "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg";
    const alt2 = "movie";
    const component = rerender.create(<PosterPreview isOpen={isOpen}
                                                     onClose={onClose}
                                                     src={src2}
                                                     alt={alt2}
                                                     fallback={fallback} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
