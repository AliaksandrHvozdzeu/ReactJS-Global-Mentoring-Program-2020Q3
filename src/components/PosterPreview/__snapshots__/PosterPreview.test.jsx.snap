// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when PosterPreview then snapshot created 1`] = `
<div
  className="poster-preview-overlay"
>
  <div
    className="poster-preview-body"
  >
    <section
      className="congratulation-poster"
    >
      <button
        aria-label="close"
        className="close-button"
        onClick={[Function]}
        type="button"
      />
      <img
        alt="movie poster logo"
        className="preview-poster-image"
        onError={[Function]}
        role="presentation"
        src="/images/not-found.png"
      />
    </section>
  </div>
</div>
`;

exports[`when PosterPreview when scr is not default 1`] = `
<div
  className="poster-preview-overlay"
>
  <div
    className="poster-preview-body"
  >
    <section
      className="congratulation-poster"
    >
      <button
        aria-label="close"
        className="close-button"
        onClick={[Function]}
        type="button"
      />
      <img
        alt="movie"
        className="preview-poster-image"
        onError={[Function]}
        role="presentation"
        src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
      />
    </section>
  </div>
</div>
`;
